crates/turborepo-lib/src/commands/link.rs

```diff
@@ -404,15 +404,35 @@ fn add_turbo_to_gitignore(base: &CommandBase) -> Result<()> {
         writeln!(gitignore, ".turbo")?;
     } else {
         let gitignore = File::open(&gitignore_path)?;
-        let mut lines = io::BufReader::new(gitignore).lines();
-        let has_turbo = lines.any(|line| line.map_or(false, |line| line.trim() == ".turbo"));
-        if !has_turbo {
-            let mut gitignore = OpenOptions::new()
-                .read(true)
-                .append(true)
-                .open(&gitignore_path)?;
-            writeln!(gitignore, ".turbo")?;
-        }
+        let lines = io::BufReader::new(&gitignore).lines();
```

Comment:
tbh I don't know why I used a BufReader here. We should just load the entire file into memory. check if there's already a turbo line, and append accordingly

Reply:
I mean, a buffered reader is technically the correct thing to use here, so I don't see it as a bad choice, just overly-defensive.

> The default is currently 8 KB

...so in spite of it being technically correct we probably read basically everybody's `.gitignore` in full the first time. We get to choose our edge case:
- everybody pays a small bit of ceremony tax for `BufReader`.
- somebody with a massive `.gitignore` file gets memory pressure.

... neither of which are going to show up in any perf trace ever because you don't call `link` except once.

So I really only care about the correctness here.